### PR TITLE
[ET-VK] Add the missing int32 -> uint8 view_convert variant

### LINKED ISSUE #22320: [ET-VK] Missing int32 -> uint8 view_convert variant: lowers cleanly, aborts at run time
### 🐛 Describe the bug

The `view_convert` shader combo lists generate `uint8 -> int32` but not the reverse, so a graph containing an `int32 -> uint8` view_convert lowers cleanly through `VulkanPartitioner` and then aborts at run time:

```
libc++abi: terminating due to uncaught exception of type vkcompute::vkapi::Error:
Exception raised from get_shader_info at backends/vulkan/runtime/api/ShaderRegistry.cpp:54:
(it != listings_.end()) is false! Could not find ShaderInfo with name view_convert_buffer_int32_uint8
```

`backends/vulkan/runtime/graph/ops/glsl/view_convert_buffer.yaml` (and the `_texture` twin):

```yaml
        - parameter_values: [uint8, float]
        - parameter_values: [uint8, half]
        - parameter_values: [uint8, int32]     # <- present
        - parameter_values: [float, int32]     # <- no [int32, uint8]
```

**Repro.** Lower `sentence-transformers/all-MiniLM-L6-v2` with `VulkanPartitioner` and run it. The conversion arrives through the attention mask:

```python
import torch, torch.nn as nn, torch.nn.functional as F
from transformers import AutoModel
from executorch.backends.vulkan.partitioner.vulkan_partitioner import VulkanPartitioner
from executorch.exir import to_edge_transform_and_lower, EdgeCompileConfig

class Emb(nn.Module):
    def __init__(self):
        super().__init__()
        self.model = AutoModel.from_pretrained(
            "sentence-transformers/all-MiniLM-L6-v2", attn_implementation="eager")
    def forward(self, input_ids, attention_mask):
        tok = self.model(input_ids=input_ids, attention_mask=attention_mask)[0]
        mask = attention_mask.unsqueeze(-1).expand(tok.size()).float()
        summed = torch.sum(tok * mask, 1, keepdim=True)
        counts = torch.sum(mask, 1, keepdim=True).clamp(min=1e-9)
        return F.normalize((summed / counts).squeeze(1), p=2, dim=1)

ep = torch.export.export(
    Emb().eval(),
    (torch.zeros(1, 128, dtype=torch.int64), torch.ones(1, 128, dtype=torch.int64)))
prog = to_edge_transform_and_lower(
    {"forward": ep}, partitioner=[VulkanPartitioner()],
    compile_config=EdgeCompileConfig(
        _core_aten_ops_exception_list=[torch.ops.aten.linear.default])).to_executorch()
open("minilm_vulkan.pte", "wb").write(prog.buffer)
```

Lowering reports a clean 2-blob partition; `executor_runner` then aborts on device.

### Expected behavior

Either generate the variant, or have the partitioner decline the node so it stays on host. Silently accepting at AOT and aborting at run time is the worst of the three, because the failure only appears on a device.

### Fix

`view_convert` is a plain elementwise `OUT_T(t_inp[...])` cast, so the pair needs nothing beyond being listed. Verified on a Galaxy S26 Ultra (Adreno 840): with the variant generated, all-MiniLM-L6-v2 runs and its embedding matches the eager reference to **cosine 0.99999726** (max abs deviation 4.0e-04), bit-identical across 90 in-process replays.

PR: #22321

### Versions

ExecuTorch `main` @ c27baa8031. Android arm64-v8a, NDK 27.1, Galaxy S26 Ultra (Adreno 840). Python 3.10, torch 2.13.0.

cc @SS-JIA @manuelcandales @digantdesai @cbilgin

### PR DESCRIPTION
**Pull request by msluszniak**
### Summary

Fixes #22320.

The `view_convert` combo lists generate `uint8 -> int32` but not the reverse, so a graph containing an `int32 -> uint8` view_convert lowers cleanly through `VulkanPartitioner` and then aborts on device:

```
Could not find ShaderInfo with name view_convert_buffer_int32_uint8
(ShaderRegistry.cpp:54, (it != listings_.end()) is false)
```

The partitioner has no visibility into which dtype pairs were generated, so the node is accepted at AOT and the failure only appears at run time. `sentence-transformers/all-MiniLM-L6-v2` hits this through its attention mask on any graph lowered with `VulkanPartitioner`.

`view_convert` is a plain elementwise `OUT_T(t_inp[...])` cast, so the pair needs nothing beyond being listed in the buffer and texture variant lists.

### Test plan

On a Galaxy S26 Ultra (Adreno 840), arm64-v8a, NDK 27.1, `main` @ c27baa8031:

Before, `executor_runner` aborts loading the model with the error above. After, all-MiniLM-L6-v2 loads and runs, and its embedding matches the eager reference:

```
minilm  vulkan vs torch ref   max|dev| 4.0e-04   cosine 0.99999726
minilm  replay 0 vs replay 29 max|dev| 0.0       cosine 1.00000000
```

Bit-identical across 90 in-process replays (3 rounds x 30 executions), so the added cast is stable as well as correct. Median 7.76 ms/iteration against 8.38 ms for the same model on XNNPACK.

cc @SS-JIA @manuelcandales @digantdesai @cbilgin

### CODE REVIEW

File: backends/vulkan/runtime/graph/ops/glsl/view_convert_buffer.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ view_convert_buffer:
         - parameter_values: [uint8, float]
         - parameter_values: [uint8, half]
         - parameter_values: [uint8, int32]
+        - parameter_values: [int32, uint8]
```
Comment (SS-JIA) [inline backends/vulkan/runtime/graph/ops/glsl/view_convert_buffer.yaml]: [High] This routes int32-to-bool storage through the shader's numeric cast. Nonzero inputs such as `2` remain `2`, while PyTorch bool conversion requires every nonzero value to become `1`. Please booleanize explicitly and test negative, zero, and non-unit positive values for both buffer and texture variants.

_AI reviewer note: this comment was generated by an AI reviewer._

**Review by SS-JIA — COMMENTED** [review]
AI-assisted review. Inline findings below.

**Review by SS-JIA — CHANGES_REQUESTED** [review]

**Review by SS-JIA — APPROVED** [review]
